tests/preprocessing/test_shadow_detection.py

```diff
@@ -0,0 +1,41 @@
+import os
+import numpy as np
+
+from pathlib import Path
+from skimage.filters import threshold_otsu
+
+from dhdt.generic.mapping_io import read_geo_image
+from dhdt.preprocessing.image_transforms import mat_to_gray
+from dhdt.preprocessing.shadow_transforms import (apply_shadow_transform,
+    _get_shadow_transforms)
+from dhdt.testing.classification_tools import calculate_accuracy
+
+TEST_DATA = os.path.join(Path(__file__).parents[2],
+                        'testdata/ShadowImages/shadow-5cvx')
+
+def _get_shadow_test_imagery():
+    I = read_geo_image(TEST_DATA + '.jpg')[0]
+    I = mat_to_gray(I)
+    M = read_geo_image(TEST_DATA + '.png')[0]
+    return I, M
+
+# testing functions, based on imagery from:
+# https://www3.cs.stonybrook.edu/~minhhoai/projects/shadow.html
+
+def test_shadow_transforms():
+    methods = list(_get_shadow_transforms().keys())
+    methods = list(methods[i] for i in [0, 5, 6, 7, 8, 10, 11, 12])
+
+    I, M = _get_shadow_test_imagery()
+    for method in methods:
```

Comment:
Here you are running the same test code on a list of parameters. Using a for loop works perfectly fine, so no need to change anything here, but just so that you know: you can achieve the same result with a pytest fixture using parameters (see [here](https://docs.pytest.org/en/7.1.x/how-to/parametrize.html)). The advantage of the latter approach is that you get a separate test for each of the parameter, so that it is then easier to identify potentially failing tests from the output (with the for loop, you will only see the overall test failing and it might be a bit trickier to find out which is(are) the problematic values).